controllers/datafoundation.go

```diff
@@ -203,26 +203,22 @@ func (r *dataFoundationReconciler) reconcilePhases() (ctrl.Result, error) {
 func (r *dataFoundationReconciler) parseSpec(offering *v1alpha1.ManagedFusionOffering) error {
 	r.Log.Info("Parsing ManagedFusionOffering Data Foundation spec")
 
+	var dfSpec dataFoundationSpec
+	err := yaml.Unmarshal([]byte(offering.Spec.Config), &dfSpec)
+	if err != nil {
+		r.Log.Error(err, "an error occurred while unmarshaling the YAML config")
+		return fmt.Errorf("failed to unmarshal YAML config")
+	}
```

Comment:
```suggestion
	if err := yaml.Unmarshal([]byte(offering.Spec.Config), &dfSpec); err != nil {
		r.Log.Error(err, "an error occurred while unmarshaling the YAML config")
		return fmt.Errorf("failed to unmarshal YAML config")
	}
```

Reply:
updated

Reply:
@bindrad if the field is present in the yaml string but it's value is empty or an invalid integer it throws an error
if field itself is not present it defaults to 0